eventuals/interrupt.h

```diff
@@ -50,56 +49,32 @@ class Interrupt final {
 
     Interrupt* interrupt_ = nullptr;
     Callback<void()> callback_;
-    Handler* next_ = nullptr;
   };
 
   Interrupt()
-    : handler_(this, []() {}) {}
+    : placeholder_handler_(this, []() {}) {}
 
-  bool Install(Handler* handler) {
-    CHECK(handler->next_ == nullptr) << "handler is already installed";
-
-    handler->next_ = head_.load(std::memory_order_relaxed);
-
-    do {
-      // Check if the interrupt has already been triggered.
-      if (handler->next_ == nullptr) {
-        return false;
-      }
-    } while (!head_.compare_exchange_weak(
-        handler->next_,
-        handler,
-        std::memory_order_release,
-        std::memory_order_relaxed));
-
-    return true;
+  [[nodiscard]] bool Install(Handler* handler) {
+    auto* stored = handler_.load();
+    CHECK(stored) << "Interrupt has already being triggered";
+    return handler_.compare_exchange_weak(stored, handler);
```

Comment:
We want this to return `false` if the interrupt has been triggered and we are not able to install the handler otherwise `true` after we've installed the handler.